docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+Type `i` to place the editor into `INSERT` mode.
+
+![submit_pr_squash_commits_4.png](./imgs/submit_pr_squash_commits_4.png)
+
+I want to keep the commit message from the first commit, so I will leave the word `pick` in front of the first commit, and change the words from `pick` to `squash` for the second and third commits. (Use the `arrow` keys to move around, and the `delete` or `backspace` key for deleting characters)
+
+![submit_pr_squash_commits_5.png](./imgs/submit_pr_squash_commits_5.png)
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+![submit_pr_squash_commits_6.png](./imgs/submit_pr_squash_commits_6.png)
+
+The `vim` editor comes up again to give me a chance to change my commit message.
+You'll notice that because I previously chose `squash` for the second and third commits, it appended the commit messages of the second and third commits onto the commit message of the first.
+
+![submit_pr_squash_commits_7.png](./imgs/submit_pr_squash_commits_7.png)
+
+This can give a user a chance to view all messages before forumating one final message.
+
+If I had a large number of commits and didn't want all commit messages to be appended together, I would have specified: `pick, fixup, fixup` instead of `pick, squash, squash` . The command `fixup` is like `squash`, except that it discards a commit's log message.
+
+In my case, I only like the message of the first commit, so I will delete the others.
+
+Press `i` to place the editor into `INSERT` mode, and prepare the single commit message.
+
+![submit_pr_squash_commits_8.png](./imgs/submit_pr_squash_commits_8.png)
+
+To save these changes, press the `escape` key, and type:
+
+```
+:wq
+```
+
+The interactive `rebase` command finally completes.
+
+![submit_pr_squash_commits_9.png](./imgs/submit_pr_squash_commits_9.png)
+
+
+Now it is time to push the changes from the local feature branch to the remote feature branch.
+
+Type :
+
+```
+git push -f
+```
+
+You can confirm that there is only 1 commit by typing:
+
+```
+git log
+```
+
+![submit_pr_squash_commits_11.png](./imgs/submit_pr_squash_commits_11.png)
+
+or via your browser in your remote feature branch.
+
+![submit_pr_squash_commits_10.png](./imgs/submit_pr_squash_commits_10.png)
+
+
+### 17. Creating the pull request (PR)
+
+Now that you have completed the `rebase` and `commit squashing` steps, it is finally time to create the pull request.
+
+In your browser, navigate to your remote `feature` branch, and press the `Compare & pull request` button.
+
+![submit_pr_create_pr_1.png](./imgs/submit_pr_create_pr_1.png)
+
+The short and long description fields of the pull request are auto-filled using the short and long description of your commit.
+
+![submit_pr_create_pr_2.png](./imgs/submit_pr_create_pr_2.png)
+
+Fill in as much extra information as possible to properly describe the purpose of the pull request ( for example, fixing a specific bug or implementing an extension point, etc).
```

Comment:
There's also a PR template that has some checkmarks and other info (although I don't know if it appears when a PR is made from a fork vs. branch). If it does show up, maybe explain to them when to check off something like `API Documentation in code was updated`?

Reply:
yeah, I was debating that when I was creating some screenshots. I decided not to in order to avoid any extra steps, but perhaps I will briefly mention this.

Reply:
@nabdelgadir , I addressed this in my latest commit.